tfhe/tests/zk_wasm_x86_test.rs

```diff
@@ -0,0 +1,84 @@
+//! Test compatibility between x86_64 and wasm proofs
+//!
+//! - Generate a crs and public key from rust
+//! - Load them in js, encrypt and prove some ciphertexts
+//! - Load the proven ciphertexts in rust and verify the proof
+
+#![cfg(feature = "zk-pok")]
+#![cfg(feature = "integer")]
+
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tfhe::safe_serialization::{safe_deserialize, safe_serialize};
+use tfhe::zk::{CompactPkeCrs, CompactPkePublicParams};
+use tfhe::{ClientKey, CompactPublicKey, ConfigBuilder, ProvenCompactCiphertextList};
+
+const SIZE_LIMIT: u64 = 1024 * 1024 * 1024;
+
+fn gen_key_and_crs() -> (CompactPublicKey, CompactPkeCrs) {
+    println!("Generating keys");
+    let config = ConfigBuilder::default().build();
+    let client_key = ClientKey::generate(config);
+    let pub_key = CompactPublicKey::new(&client_key);
+
+    println!("Generating crs");
+    let crs = CompactPkeCrs::from_config(config, 16).unwrap();
+
+    (pub_key, crs)
+}
+
+fn gen_proven_ct_in_wasm(path: &Path) {
+    println!("Generating proven ciphertext in wasm");
+    let mut child = Command::new("node")
+        .arg("index.js")
+        .current_dir(path)
+        .spawn()
+        .expect("Failed to run node script");
+
+    let exit_status = child.wait().unwrap();
+    if let Some(exit_code) = exit_status.code() {
+        if exit_code == 0 {
+            return;
+        }
+    }
+
+    panic!("node script returned a non-0 code.");
+}
+
+fn verify_proof(
+    public_key: &CompactPublicKey,
+    crs: &CompactPkePublicParams,
+    proven_ct: &ProvenCompactCiphertextList,
+) {
+    println!("Verifying proof");
+    match proven_ct.verify(crs, public_key, &[]) {
+        tfhe::zk::ZkVerificationOutCome::Valid => {}
```

Comment:
No log message to say that verification has passed ?

Reply:
good idea !